optimum/intel/openvino/modeling_base.py

```diff
@@ -70,9 +70,16 @@ def __init__(
         self.is_dynamic = dynamic_shapes
         self.ov_config = {} if ov_config is None else {**ov_config}
         self.preprocessors = kwargs.get("preprocessors", [])
+        self.compile_only = kwargs.get("compile_only", False)
         enable_compilation = kwargs.get("compile", True)
 
-        if self.is_dynamic:
+        if self.compile_only and not enable_compilation:
+            raise ValueError(
+                "`compile_only` mode does not support disabling compilation."
+                "Please provide `compile=True` if you want to use `compile_only=True` or set `compile_only=False`"
+            )
+
+        if self.is_dynamic and not self.compile_only:
```

Comment:
if I understand correctly, `compile only` also doesn't allow re-compilation, so the argument `compile=` doesn't make much sense when used with `compile_only`, whether it's true (re-compilation) or false (disable compilation) ?

Reply:
yes, recompilation is not supported, but we need to keep compile flag for regular case.